### PR TITLE
plugin/reload: Add metrics

### LINKED ISSUE #2920: plugin/reload: add failed reload counter
We should add a failed reload counter metrics to detect non-valid config being pushed to coredns. If coredns crashes or gets restarted and invalid config is put on disk it will fail to start. Having a metrics will give oncallers some change to preempt this failure mode.

### PR DESCRIPTION
**Pull request by sharmaansh21**
### 1. Why is this pull request needed and what does it do?
Fixes #2920

### CODE REVIEW

**Comment by sharmaansh21** [comment]
@miekg please let me know if something is missed or I can improve in this.

File: plugin/reload/metrics.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics for the reload plugin
+var (
+	FailedCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "reload",
+		Name:      "failed_count",
+		Help:      "A metrics that counts the number of failed reload attempts.",
```
Comment (ihac) [inline plugin/reload/metrics.go]: `metrics` -> `metric`

Comment (sharmaansh21) [reply plugin/reload/metrics.go]: Fixed  thanks :sweat_smile:

File: plugin/reload/README.md [hunk 2]
```diff
@@ -92,6 +92,12 @@ In general be careful with assigning new port and expecting reload to work fully
 Also any `import` statement is not discovered by this plugin. This means if any of these imported files
 changes the *reload* plugin is ignorant of that fact.
 
+## Metrics
+
+ If monitoring is enabled (via the *prometheus* directive) then the following metric is exported:
+
+* `coredns_reload_failed_count{}` - counts the number of failed reload attempts.
```
Comment (miekg) [inline plugin/reload/README.md]: this needs to be `coredns_reload_failed_count_total` iirc

File: plugin/reload/metrics.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Metrics for the reload plugin
+var (
+	FailedCount = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "reload",
+		Name:      "failed_count",
+		Help:      "A metric that counts the number of failed reload attempts.",
```
Comment (miekg) [inline plugin/reload/metrics.go]: Help: Counter of the number of failed reload attempts.

**Review by miekg — COMMENTED** [review]
thank for picking this up; some naming comments

File: plugin/reload/metrics.go (hunk 3 above)
Comment (sharmaansh21) [reply plugin/reload/metrics.go]: Fixed

File: plugin/reload/README.md (hunk 2 above)
Comment (sharmaansh21) [reply plugin/reload/README.md]: Fixed

**Pull request merged**